suite/tests/client-interface/emulation_api_simple.dll.c

```diff
@@ -64,16 +88,239 @@ dr_client_main(client_id_t id, int argc, const char *argv[])
 
     if (!drmgr_register_bb_app2app_event(event_instruction_change, NULL))
         DR_ASSERT(false);
+    if (!drmgr_register_bb_instrumentation_event(event_analysis, event_insertion, NULL))
+        DR_ASSERT(false);
+    derived_marker = drmgr_reserve_note_range(1);
+    DR_ASSERT(derived_marker != DRMGR_NOTE_NONE);
 }
 
 static void
 event_exit(void)
 {
-    if (!drmgr_unregister_bb_app2app_event(event_instruction_change))
+#if VERBOSE
+    dr_fprintf(STDERR, "Found and emulated %d instructions fully, %d partly\n",
```

Comment:
Maybe add this to expected output pattern?

Reply:
This is running hello,world which is non-deterministic across compilers and machines so printing this would make the test flaky.

It would have to be a pure-asm app.  I kind of prefer hello,world since it has hundreds of the partly-emulated and dozens of fully-emulated.

Maybe some sanity check that at least some of each occurred would be good.  I added an assert that they are both > 0.

Reply:
> Maybe some sanity check that at least some of each occurred would be good. I added an assert that they are both > 0.

Yes, this will help. I have found atleast 3 tests (one recently) where the test wasn't working as intended because the to-be-tested path wasn't being invoked at all. So I think having such asserts will be good in general.

Reply:
Done.